src/lib/dialog/dialog.ts

```diff
@@ -52,6 +48,34 @@ export class MdDialog {
       @Optional() @SkipSelf() private _parentDialog: MdDialog) { }
 
   /**
+   * Opens a modal dialog containing the given TemplateRef.
+   * @param TemplateRef to load into the dialog
+   * @param config Extra configuration options.
+   * @returns Reference to the newly-opened dialog.
+   */
+  openFromTemplateRef<T>(templateRef: TemplateRef<T>, config?: MdDialogConfig): MdDialogRef<T> {
```

Comment:
I think this method should lose the generic and return `MdDialogRef<void>`. since the generic refers to the type of `componentInstance`.